common/models/model-method.js

```diff
@@ -10,4 +12,53 @@ module.exports = function(ModelMethod) {
    * @class ModelMethod
    * @inherits WorkspaceEntity
    */
+  app.once('ready', function() {
+    ready(ModelMethod);
+  });
 };
+
+
+function ready(ModelMethod) {
+  var debug = require('debug')('workspace:model-method');
+  var PackageDefinition = app.models.PackageDefinition;
+  var ModelDefinition = app.models.ModelDefinition;
+  
+  PackageDefinition.observe('after save', function getPackageDef(ctx, next) {
+    var pkg = ctx.instance ? ctx.instance.toObject() : ctx.data;
+
+    if (!pkg) {
+      debug('Cannot find Loopback version, defaulting to 2.x.');
+    } else {
+      var version = pkg.dependencies['loopback'];
+      var range = semver.Range(version);
+       // pkg.devDependencies['loopback'] = '1.x';
+      // pkg.optionalDependencies['loopback'] = '1.x';
+      debug('Loopback version: %s', version);
+    
+      if (semver.satisfies('3.0.0', range)) {
+        ModelMethod.observe('before save', function injectPrototypeName(ctx2, next) {
+          var Entity = ctx2.Model;
+          var properties = Entity.definition.properties;
+          var data = ctx2.instance ? ctx2.instance.toObject() : ctx2.data;
+
+          if (!('isStatic' in properties || 'isStatic' in data)) {
+            return next();
+          }
+          
+          var isStatic = data.isStatic;
+          if (isStatic !== undefined) {
+            var name = data.name;
+            var matchName = name.match(/^prototype\.(.*)$/);
+            if (!isStatic && (matchName === null || !matchName)) {
+              (ctx2.instance || ctx2.data).name = 'prototype.' + name;
+            }
+          }
+          // delete properties.isStatic;
+          delete ctx2.instance.__data.isStatic;
```

Comment:
1) If you want to unset a property, you should call `ctx2.instance.unsetAttribute('isStatic')`.

2) I find this implementation rather hacky. If I am reading the code correctly, it will leave `ctx.instance` in a different state that no longer matches the API contract.